File: Model/Contato.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AgendaAngular.Model
{
    public enum Categoria
    {
        Casa = 1,
        Trabalho = 2,
        Outro = 3
    }

    public class Contato
    {
        public int Id { get; set; }

        public string Nome { get; set; }

        public virtual List<Email> EmailList { get; set; }

        public virtual List<Telefone> TelefoneList { get; set; }

        public string Endereco { get; set; }

        public string Empresa { get; set; }

    }
}

File: ClientApp/app/classes/contato.ts
export interface Contato {
    id: number;
    nome: string;
    emailList: Email[];
    telefoneList: Telefone[];
    endereco: string;
    empresa: string;
}

export interface Email {
    id: number;
    endereco: string;
    contatoId: number;
    categoria: Categoria;
}

export interface Telefone {
    id: number;
    numero: string;
    contatoId: number;
    categoria: Categoria;
}

export enum Categoria {
    Casa = 1,
    Trabalho = 2,
    Outro = 3
}